realm/realm-library/src/main/java/io/realm/RealmInteger.java

```diff
@@ -16,31 +16,32 @@
 package io.realm;
 
 import io.realm.annotations.Beta;
+import io.realm.internal.ManagableObject;
+import io.realm.internal.counters.UnmanagedRealmInteger;
 
 
 /**
  * TODO: document the behaviour of this object, as soon as the implementation defines it.
  */
 @Beta
-public final class RealmInteger extends Number implements Comparable<RealmInteger> {
-    private long value;
+public abstract class RealmInteger extends Number implements Comparable<RealmInteger>, ManagableObject {
 
     /**
      * Creates a new {@code RealmInteger}, with the specified initial value.
      *
      * @param value initial value.
      */
-    public RealmInteger(long value) {
-        this.value = value;
+    public static RealmInteger valueOf(long value) {
```

Comment:
It would be needed for e.g creating initial values in model classes, just like we have an unmanaged RealmList:

Example:
```
public class Party extends RealmModel {
  private RealmInteger guests = RealmInteger.valueOf(1); 
}
```